opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -0,0 +1,274 @@
+        timestamp: typing.Optional[int] = None,
+    ) -> None:
+        """Adds an `Event`.
+        Adds a single `Event` with the name, an event formatter that calculates
+        the attributes lazily and, optionally, a timestamp. Implementations
+        should generate a timestamp if the `timestamp` argument is omitted.
+        """
+
+    @abc.abstractmethod
+    def update_name(self, name: str) -> None:
+        """Updates the `Span` name.
+
+        This will override the name provided via :func:`opentelemetry.trace.Tracer.start_span`.
+
+        Upon this update, any sampling behavior based on Span name will depend
+        on the implementation.
+        """
+
+    @abc.abstractmethod
+    def is_recording_events(self) -> bool:
+        """Returns whether this span will be recorded.
+
+        Returns true if this Span is active and recording information like
+        events with the add_event operation and attributes using set_attribute.
+        """
+
+    @abc.abstractmethod
+    def set_status(self, status: Status) -> None:
+        """Sets the Status of the Span. If used, this will override the default
+        Span status, which is OK.
+        """
+
+    def __enter__(self) -> "Span":
+        """Invoked when `Span` is used as a context manager.
+
+        Returns the `Span` itself.
+        """
+        return self
+
+    def __exit__(
+        self,
+        exc_type: typing.Optional[typing.Type[BaseException]],
+        exc_val: typing.Optional[BaseException],
+        exc_tb: typing.Optional[python_types.TracebackType],
+    ) -> None:
+        """Ends context manager and calls `end` on the `Span`."""
+
+        self.end()
+
+
+class TraceFlags(int):
+    """A bitmask that represents options specific to the trace.
+
+    The only supported option is the "sampled" flag (``0x01``). If set, this
+    flag indicates that the trace may have been sampled upstream.
+
+    See the `W3C Trace Context - Traceparent`_ spec for details.
+
+    .. _W3C Trace Context - Traceparent:
+        https://www.w3.org/TR/trace-context/#trace-flags
+    """
+
+    DEFAULT = 0x00
+    SAMPLED = 0x01
+
+    @classmethod
+    def get_default(cls) -> "TraceFlags":
+        return cls(cls.DEFAULT)
+
+    @property
+    def sampled(self) -> bool:
+        return bool(self & TraceFlags.SAMPLED)
+
+
+DEFAULT_TRACE_OPTIONS = TraceFlags.get_default()
+
+
+class TraceState(typing.Dict[str, str]):
+    """A list of key-value pairs representing vendor-specific trace info.
+
+    Keys and values are strings of up to 256 printable US-ASCII characters.
+    Implementations should conform to the `W3C Trace Context - Tracestate`_
+    spec, which describes additional restrictions on valid field values.
+
+    .. _W3C Trace Context - Tracestate:
+        https://www.w3.org/TR/trace-context/#tracestate-field
+    """
+
+    @classmethod
+    def get_default(cls) -> "TraceState":
+        return cls()
+
+
+DEFAULT_TRACE_STATE = TraceState.get_default()
```

Comment:
Any reason these were moved as well? Was this to get around a circular dependency issue?

Reply:
I was thinking that the TraceState and TraceFlags classes might be better living in `trace/__init__py` since it doesn't seem to fit strictly within a `Span` but i'm mostly curious.